core/identity/src/service/identity.rs

```diff
@@ -392,6 +396,12 @@ impl IdentityService {
             let this = this.clone();
             async move { this.lock().await.update_identity(update).await }
         });
+        let this = me.clone();
+        let _ = bus::bind(model::BUS_ID, move |drop: model::Drop| {
+            let this = this.clone();
+            async move { this.lock().await.drop_identity(drop).await }
```

Comment:
We can use the mechanism of Identity Events introduced for informing subscribed services about identity (Un)Locking.